src/mbgl/tile/vector_tile.cpp

```diff
@@ -15,9 +15,17 @@ class VectorTileLayer;
 
 using packed_iter_type = protozero::iterator_range<protozero::pbf_reader::const_uint32_iterator>;
 
+struct VectorTileLayerData {
```

Comment:
> Why does the VectorTileFeature need a std::shared_ptr<VectorTileLayerData> layerData rather than just const VectorTileLayer&. This indicates to me that somewhere in the code expects to keep a VectorTileFeature instance alive longer than the parent VectorTileLayer

@springmeyer Yeah -- this is a possibility that (I think) did not exist before, but was introduced in this PR.  (And apologies: I should have left a comment justifying the `shared_ptr`s introduced here.) 

The reason it can now happen is over in `SymbolLayout`, and specifically `SymbolFeature` ([here](https://github.com/mapbox/mapbox-gl-native/pull/7939/files#diff-9a0baf47fcfbcbe492c348bfe28cacefL11)).  `SymbolLayout` is constructed with a vector tile layer, and then later on--when there's no guarantee that the vector tile data is still around--expected to produce a "bucket" containing final rendering data.  Before this PR, we copied the necessary data into `SymbolFeature` structs which `SymbolLayout` could hang on to; but because we need feature properties for data-driven styling, `SymbolFeature` now holds a reference to the `VectorTileFeature` because it turns out that copying those over is a large memory burden.

Because of this, `VectorTileFeature`s may need to outlive their parent `VectorTileLayer`s, so it needs shared ownership of `VectorTileLayerData`.

As for the `shared_ptr<const std::string> data` that's held by `VectorTileLayerData` -- I think we need this basically as a consequence of the above, since `VectorTileFeature`s may parse PBF data lazily.  But even if we didn't, it's making explicit a dependency that already existed, since there's no explicit guarantee that the parent `VectorTileData` would outlive a particular `VectorTileLayer`.  The lifecycles _happened_ to be okay before this, but only because of implicit constraints on how we used these objects.

@jfirebaugh should probably check my math on this, as I haven't spent time in the parts of our vector tile parsing/caching stuff outside of `SymbolLayout`.